fitbenchmarking/fitbenchmarking
# Updated linting to ruff

## Description of work

Fixes #1303

- [x] The linting now uses `Ruff`.
- [x] `pylint`, `flake8` and `isort`  have been removed from the dev dependencies or precommit hooks.
- [x] Ruff rule set is finalised as follows:
```
"E", # pycodestyle
"F", # Pyflakes
"UP", # pyupgrade
"I", # isort
"NPY", # Numpy
"PERF", # Perflint
"LOG", # flake8-logging
"ICN", # flake8-import-conventions
"SIM", # flake8-simplify
```

**Note:** There are more rule sets that will be useful to add down the line. They are not added in this PR to limit the scope but I will create some issues for those rule sets.




## Testing Instructions



1. run `pip install ruff` or `pip install --upgrade .[dev]` in an existing environment or `pip install .[dev]` in a new env.
2. run `ruff check .`. It should say **All checks passed!**. But incase of errors,  run `ruff check --fix` to fix the fixable mistakes. The rest will have to be manually fixed.
3. run `ruff format --check`. It should say **153 files already formatted**. But incase of formatting errors, run `ruff format .`

## Checklist



      - #1326 
      - #1327
      - #1328
      - #1329
      - #1330
